libraries/chain/block_state.cpp

```diff
@@ -26,7 +26,7 @@ block_state::block_state(const block_header_state& bhs, deque<transaction_metada
 {
    block->transactions = std::move(trx_receipts);
 
-   if( qc && bhs.is_needed(*qc) ) {
+   if( qc ) {
```

Comment:
Why is `is_needed` removed? We now always include QC block extension?

Reply:
> Why is `is_needed` removed? We now always include QC block extension?

The parameter passed is a `std::optional<quorum_certificate>`, and the optional is empty when the qc should not be included. Whenever a qc is passed, it should always be included. 
With the check in this branch, the test doesn't pass.